neurolib/models/multimodel/builder/base/neural_mass.py

```diff
@@ -0,0 +1,200 @@
+        # used in determining portion of the full system's state vector
+        self.idx_state_var = None
+        self.initialised = False
+
+        # initialise possible helpers
+        self.helper_symbols = {symbol: se.Symbol(symbol) for symbol in self.helper_variables}
+
+        # initialise possible callback functions
+        self.callback_functions = {function: se.Function(function) for function in self.python_callbacks}
+
+        self._validate_parameters()
+
+    def __str__(self):
+        """
+        String representation.
+        """
+        return (
+            f"Neural mass: {self.name} with {self.num_state_variables} "
+            f"state variables: {', '.join(self.state_variable_names)}"
+        )
+
+    def describe(self):
+        """
+        Return description dict.
+        """
+        return {field: getattr(self, field) for field in self.DESCRIPTION_FIELD}
+
+    def _initialize_state_vector(self):
+        """
+        Initialize state vector. By default it is all zeroes.
+        """
+        self.initial_state = [0.0] * self.num_state_variables
+
+    def _validate_parameters(self):
+        """
+        Validate parameters - check if self.parameters contains all required
+        parameters.
+        """
+        assert all(key in self.parameters for key in self.required_parameters), set(self.required_parameters) - set(
+            self.parameters.keys()
+        )
+
+    def _validate_callbacks(self, callback_list):
+        """
+        Validate callbacks - mainly the length and symbolic function names.
+
+        :param callback_list: list of callbacks
+        :type callback_list: list[tuple|list]
+        """
+        assert len(callback_list) == len(self.callback_functions)
+        assert all(
+            callback_from_list[0] == callback_from_init
+            for callback_from_list, callback_from_init in zip(callback_list, self.callback_functions.values())
+        )
+        assert all(callable(callback[1]) for callback in callback_list)
+
+    def init_mass(self, start_idx_for_noise=None):
+        """
+        Initialise neural mass. Usually just initialise the state vector,
+        possibly can be subclassed and do other initialisation.
+        """
+        self._initialize_state_vector()
+        assert self.index is not None
+        if start_idx_for_noise is None:
+            start_idx_for_noise = self.index
+        if self.noise_input_idx is None:
+            self.noise_input_idx = [start_idx_for_noise + i for i in range(self.num_noise_variables)]
+        self.initialised = True
+
+    def update_parameters(self, parameters_dict):
+        """
+        Update parameters of the mass.
+
+        :param parameters_dict: new parameters for this mass
+        :type parameters_dict: dict
+        """
+        assert isinstance(parameters_dict, dict)
+        self.parameters.update(parameters_dict)
+        # validate again
+        self._validate_parameters()
+
+    def _callbacks(self):
```

Comment:
ignore `callbacks` and `numba_callbacks` for now - they are important for AdEx but not otherwise